crates/core/tedge_actors/src/message_boxes.rs

```diff
@@ -142,44 +143,46 @@ pub trait MessageBox: 'static + Sized + Send + Sync {
     fn logging_is_on(&self) -> bool;
 }
 
+enum WrappedInput<Input> {
+    Message(Input),
+    RuntimeRequest(RuntimeRequest),
+}
+
+// TODO: rename
+#[async_trait]
+trait Foo<Input> {
+    async fn a(&mut self) -> Result<Option<Input>, RuntimeRequest>;
```

Comment:
```suggestion
    async fn try_recv(&mut self) -> Result<Option<WrappedInput>, ChannelError>;
```
Else how will we propagate any channel specific errors? For those actors not wanting to handle the `RuntimeRequest`, the `c` variant is already there. So, I don't see any issues in this being wrapped.

Reply:
The point of this fn is the ergonomics of handling the RuntimeRequests. So I think Result<_, Either<ChannelError, RuntimeRequest> might be better.

CombineReceiver never produces Channel errors so I guess there's some message box impl where this would be useful, could you point me to an example where this would be useful?

Reply:
Okay I just noticed that the `recv` method does not return any `ChannelError` but only an `Option`. It was only `send` that returns `ChannelError`. I guess any errors while `recv`ing from a channel is implicitly mapped into `Option:None` instead of returning that error. So, this signature is fine.